app/scripts/models/auth_brokers/redirect.js

```diff
@@ -76,7 +76,7 @@ define([
       // event handlers before the flow completes.
       var self = this;
       return p().delay(100).then(function () {
-        if (self.isOriginalTab()) {
+        if (self.isOriginalTab() || account.get('sessionToken') && alwaysRedirect) {
```

Comment:
Passing in `alwaysRedirect` feels janky and like an implementation detail the view doesn't need to know about. The broker already has access to the `relier`, so the flag can be fetched by the broker directly from the relier. 

> Any better way to do this, is there anything else we can use instead of sessionToken to signal that this is SAME_BROWSER?

1. We need to know this already in app-start - see https://github.com/mozilla/fxa-content-server/blob/4898dc006863fe26746b7562cc5dad78c36ca2f5/app/scripts/lib/app-start.js#L501
1. We have a notion of "verification info models" in https://github.com/mozilla/fxa-content-server/tree/master/app/scripts/models/verification. Logically, knowing whether the user is verifying in the same browser seems like it is "verification info". Maybe we can add a bit onto the verification info models and pass the model in to the broker?